lib/src/common/parameters/excluded_identifiers_list_parameter.dart

```diff
@@ -33,12 +33,34 @@ class ExcludedIdentifiersListParameter {
     );
   }
 
+  /// Method for creating from json data with default params
+  factory ExcludedIdentifiersListParameter.defaultFromJson(
+    Map<String, dynamic> json,
+  ) {
+    final exclude = <ExcludedIdentifierParameter>[];
+
+    final excludeList =
+        json[ExcludedIdentifiersListParameter.excludeParameterName]
+                as Iterable? ??
+            [];
+
+    for (final item in excludeList) {
+      if (item is Map) {
+        exclude.add(ExcludedIdentifierParameter.fromJson(item));
+      }
+    }
+    return ExcludedIdentifiersListParameter(
+      exclude: exclude,
+    );
+  }
+
   /// Returns whether the target node should be ignored during analysis.
   bool shouldIgnore(Declaration node) {
     final methodName = node.declaredElement?.name;
 
-    final excludedItem =
-        exclude.firstWhereOrNull((e) => e.methodName == methodName);
+    final excludedItem = exclude.firstWhereOrNull(
+      (e) => e.methodName == methodName || e.className == methodName,
```

Comment:
This removes some confusion.

And also highlights that we may not be comparing what we think we should be comparing.

We should check whether the declaration is a class before running `exclude.className == declarationName`; same for method names.

```suggestion
      (e) => e.methodName == declarationName || e.className == declarationName,
```